src/lib.rs

```diff
@@ -759,11 +775,27 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
 /// # Returns
 ///
 /// Name to use for task.
-fn lookup_task_name(is_enterprise: bool, task_name: &str, platform: &str) -> String {
+fn lookup_task_name(
+    is_enterprise: bool,
+    task_name: &str,
+    platform: &str,
+    unique_gen_suffix: Option<String>,
```

Comment:
Is it possible change this to Option<&str>? I think you can deref in the calling code that invokes this function. String is generally when you want to own it rather than just reference it (but I haven't used rust in a while so maybe it's necessary here)